spec/EventListener/ShippingExportEventListenerSpec.php

```diff
@@ -1,75 +1,74 @@
-        $shippingExport->setExportedAt($date)->shouldBeCalled();
-
-        $exportShipmentEvent->getSubject()->willReturn($shippingExport);
-        $shippingExport->getShippingGateway()->willReturn($shippingGateway);
-        $shippingExport->getShipment()->willReturn($shipment);
-
-        $webClient->createShipment($shipment)->willReturn(['id' => 10]);
-
-        $webClient->getShipmentById(10)->willReturn(['status' => WebClientInterface::CONFIRMED_STATUS, 'id' => 10]);
-        $webClient->getLabels()->willReturn('Label content');
-        $shipment->getOrder()->willReturn($order);
-        $order->getNumber()->willReturn(1);
-        $this->exportShipment($exportShipmentEvent);
-    }
-
-}
\ No newline at end of file
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace spec\BitBag\SyliusInPostPlugin\EventListener;
+
+use BitBag\SyliusInPostPlugin\Api\WebClientInterface;
+use BitBag\SyliusInPostPlugin\EventListener\ShippingExportEventListener;
+use BitBag\SyliusShippingExportPlugin\Entity\ShippingExportInterface;
+use BitBag\SyliusShippingExportPlugin\Entity\ShippingGatewayInterface;
+use Doctrine\Persistence\ObjectManager;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+final class ShippingExportEventListenerSpec extends ObjectBehavior
+{
+    function let(ObjectManager $manager, FlashBagInterface $flashBag, WebClientInterface $webClient, Filesystem $filesystem): void
+    {
+        $shippingLabelsPath = 'labels';
+        $this->beConstructedWith($manager, $flashBag, $webClient, $filesystem, $shippingLabelsPath);
+    }
+
+    function it_is_initializable(): void
+    {
+        $this->shouldHaveType(ShippingExportEventListener::class);
+    }
+
+    function it_exports_shipment(
+        ResourceControllerEvent $exportShipmentEvent,
+        ShippingGatewayInterface $shippingGateway,
+        ShipmentInterface $shipment,
+        WebClientInterface $webClient,
+        ShippingExportInterface $shippingExport,
+        Order $order
+    ): void {
+        $shippingGateway->getCode()->willReturn(ShippingExportEventListener::INPOST_SHIPPING_GATEWAY_CODE);
+        $shippingGateway->getConfigValue('wsdl')->willReturn('wsdl');
+
+        $webClient->setShippingGateway($shippingGateway)->shouldBeCalled();
+        $webClient->createShipment($shipment)->shouldBeCalled();
+        $shippingExport->setExternalId('10')->shouldBeCalled();
```

Comment:
Are you sure that all of these calls which you check with `shouldBeCalled` call are crucial to the export process?
If yes, then leave it as is, but remember to use it only if needed because extensive use of this method may lead to the situation, that almost every change in the code will break the tests

Reply:
Imo they're crucial enough to leave it like it is. If anyone changes the code, he should also make sure that the process that is provided by these methods is to be maintained.

Reply:
Ok